doc/secure.md

```diff
@@ -0,0 +1,88 @@
+
+### Wallet
+
+[Encrypting the Wallet](/managing-wallets.md#encrypting-the-wallet)
+
+[Backing Up the Wallet](/managing-wallets.md#backing-up-the-wallet)
+
+Do not buy _wallet.dat_ files from anywhere. There are several websites that sell such files to newbies. You can read more about the issue here: https://github.com/bitcoin-core/gui/issues/77#issuecomment-721927922
+
+Understanding the difference between legacy and descriptor wallets:
+
+There is no XPUB in legacy wallets, `dumpwallet` will return private keys. `listdescriptors` can be used to get XPUB and XPRV for descriptor wallets.
+
+
+### Other
+
+*notify config and commandline options accept shell commands to be based on different events in Bitcoin Core. Be careful while using these options and ensure that the commands are safe either by manual inspection or restrictions/alerts in the system used.
+
+## Developers
+
+ℹ️ It’s a myth that those break cannot build.
+
+- [Developer Notes](/developer-notes.md)
+
+### Secure Programming Practices in C++
+
+**Undefined Behavior (UB)**: there are no restrictions on the behavior of the program if certain rules of the language are violated. Examples of undefined behavior are uninitialized scalars, memory access outside of array bounds, signed integer overflow, null pointer dereference, access to an object through a pointer of a different type, etc. Compilers are not required to handle undefined behavior (although many simple situations are diagnosed), and the compiled program is not required to do anything meaningful.
+
+**Uninitialized variables** are Undefined Behavior, a typical outcome of which is reading stale stack or register values. In the best case, it generates a pseudorandom result which in edge cases might be the expected result or fixed by the compiler. In the worst case, an exploit may be created that leaks secrets by allowing use of an attacker-controlled value.
+
+**MemorySanitizer** (MSan) is a tool that detects uninitialized reads. It consists of a compiler instrumentation module and a run-time library. Typical slowdown is 3x and memory usage is 2x. MSan is supported on Linux/NetBSD/FreeBSD and requires compiling and linking all the code in your program, including libraries it uses and the C++ standard library, with -fsanitize=memory -fPIE -pie.
+
+**Valgrind** is a debugging and profiling tool suite to make programs faster and more correct. Its most popular tool, Memcheck, can detect memory-related errors common in C and C++ programs that can lead to crashes and unpredictable behavior. Here is a tutorial. Memcheck is not perfect: typical slowdown is 3-10x, memory usage is 2x, it can produce false positives (there are mechanisms for suppressing these; see the valgrind.supp file in Bitcoin Core), and it doesn’t currently detect out-of-range reads or writes to arrays allocated statically or on the stack.
+
+### Finding vulnerabilities in pull requests during reviews
```

Comment:
The developer notes contain some remarks for reviewers. Maybe it makes sense to have a separate section there too.

Reply:
Done in https://github.com/bitcoin/bitcoin/commit/05511d08be7e306a60c758ee2a64edc0cd63478e